### PR TITLE
Updated and rebased to LLVM8

### PR DESCRIPTION
**Pull request by anton-malakhov**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Replaces #61

### CODE REVIEW

**Comment by anton-malakhov** [comment]
Appveyor:
```
CMake Error at cmake/modules/CheckCompilerVersion.cmake:40 (message):
  Host Visual Studio version should be at least 19.1 because LLVM will soon
  use new C++ features which your toolchain version doesn't support.  Your
  version is 19.0.24241.7.  You can temporarily opt out using
  LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN, but very soon your toolchain won't be
  supported.
Call Stack (most recent call first):
  cmake/modules/CheckCompilerVersion.cmake:48 (check_compiler_version)
  cmake/config-ix.cmake:13 (include)
  CMakeLists.txt:590 (include)
```

**Comment by anton-malakhov** [comment]
```
+ ../build/bin/llvm-lit -vv Transforms ExecutionEngine Analysis CodeGen/X86
llvm-lit: /usr/local/miniconda/conda-bld/llvmdev_1554170587448/work/utils/lit/lit/TestingConfig.py:102: fatal: unable to parse config file '/usr/local/miniconda/conda-bld/llvmdev_1554170587448/work/test/lit.site.cfg.py', traceback: Traceback (most recent call last):
  File "/usr/local/miniconda/conda-bld/llvmdev_1554170587448/work/utils/lit/lit/TestingConfig.py", line 89, in load_from_path
    exec(compile(data, path, 'exec'), cfg_globals, None)
  File "/usr/local/miniconda/conda-bld/llvmdev_1554170587448/work/test/lit.site.cfg.py", line 41, in <module>
    config.enable_threads = ON
NameError: name 'ON' is not defined
```
Unless somebody has bandwidth to investigate or knowledge of llvm's testing system, I'd suggest to disable it again :(

**Comment by anton-malakhov** [comment]
Azure, win:
```
    C:\ProgramData\chocolatey\bin\gcc.exe   -MD -GL -o CMakeFiles/cmTC_a7de9.dir/testCCompiler.c.obj   -c testCCompiler.c
    gcc.exe: error: unrecognized command line option '-GL'; did you mean '-L'?
```
I probably need help with this one..

**Comment by isuruf** [comment]
Ignore azure win. There are known failures with azure and win and cmake.

**Comment by anton-malakhov** [comment]
Looks like appveyor timed out though it did build everything successfully.

**Comment by anton-malakhov** [comment]
@isuruf the last commit didn't help appveyor to finish in the given time. I have no more ideas how to accelerate the process.  Please help

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ source:
   sha256: {{ sha256 }}
   patches:
     - llvm-lto-static.patch   # [win]
-    - partial-testing.patch
+    - partial-testing.patch   # [not win]
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you revert this change now that azure works and timing out is not an issue?

Comment (anton-malakhov) [reply recipe/meta.yaml]: Thanks for your help Isuru! Honestly, this patch look unnecessary without calls for llvm-lit which I also had to comment out. I'll rather comment it out completely with a TODO note

**Comment by anton-malakhov** [comment]
It occurred to me that as long we are not restricted by compile time on windows, I can try re-enable testing in a normal way as I tried for the last update. I'll return to it later, if it is urgent for someone to merge this PR as is - it's fine, you don't nee to wait for me

**Comment by isuruf** [comment]
Now, the limit is 6 hours on all platforms. I'd like to test some packages with this, so I'm going to merge. Thanks for your work on this. Please send a separate PR and I'll be happy to merge.
